src/Compilers/Core/Portable/Symbols/Attributes/UnmanagedCallersOnlyAttributeData.cs

```diff
@@ -0,0 +1,33 @@
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.Symbols;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class UnmanagedCallersOnlyAttributeData
+    {
+        internal static readonly UnmanagedCallersOnlyAttributeData Uninitialized = new UnmanagedCallersOnlyAttributeData(callingConventionTypes: null, isValid: false);
+
+        public readonly ImmutableHashSet<INamedTypeSymbolInternal>? CallingConventionTypes;
+
+#pragma warning disable 8775 // Invariant is verified in the constructor
+        [MemberNotNullWhen(true, nameof(CallingConventionTypes))]
+        public bool IsValid { get; }
```

Comment:
>IsValid [](start = 20, length = 7)

Is `IsValid` only used to for the one `Uninitialized` value? If so, it seems unnecessary since we're already using comparing references with `Interlocked.CompareExchange`.

Reply:
No, it's also used for when there was invalid data in the attribute, or `CallConvs` was specified twice.

---
In reply to: [482198657](https://github.com/dotnet/roslyn/pull/47311#discussion_r482198657) [](ancestors = 482198657)